tests/test_grid3d/test_grid_property_grdecl.py

```diff
@@ -0,0 +1,166 @@
+coordinates = st.floats(min_value=-100.0, max_value=100.0)
+increments = st.floats(min_value=1.0, max_value=100.0)
+dimensions = st.tuples(indecies, indecies, indecies)
+
+
+grids = st.builds(
+    create_grid,
+    dimension=dimensions,
+    origin=st.tuples(coordinates, coordinates, coordinates),
+    increment=st.tuples(increments, increments, increments),
+    rotation=st.floats(min_value=0.0, max_value=90),
+)
+
+
+@settings(suppress_health_check=[HealthCheck.function_scoped_fixture], deadline=None)
+@given(grids)
+def test_grid_to_from_grdecl_file_is_identity(tmp_path, grid):
+    filepath = tmp_path / "grid.grdecl"
+    grid.to_file(filepath, fformat="grdecl")
+    grid_from_file = Grid().from_file(filepath, fformat="grdecl")
+
+    assert grid.dimensions == grid_from_file.dimensions
+    assert np.array_equal(grid.actnum_array, grid_from_file.actnum_array)
+
+    for prop1, prop_from_file in zip(
+        grid.get_xyz_corners(), grid_from_file.get_xyz_corners()
+    ):
+        assert_allclose(
+            prop1.get_npvalues1d(), prop_from_file.get_npvalues1d(), atol=1e-3
+        )
+
+
+@settings(suppress_health_check=[HealthCheck.function_scoped_fixture], deadline=None)
+@given(grids)
+def test_gridprop_to_from_file_is_identity(tmp_path, grid):
+    filepath = tmp_path / "gridprop.grdecl"
+
+    for prop in grid.get_xyz_corners():
+        prop.to_file(filepath, fformat="grdecl")
+        prop_from_file = GridProperty().from_file(
+            filepath, name=prop.name, fformat="grdecl", grid=grid
+        )
+
+        assert_allclose(
+            prop.get_npvalues1d(), prop_from_file.get_npvalues1d(), atol=1e-3
+        )
+
+
+@pytest.mark.parametrize(
+    "file_data",
+    [
+        "PROP\n 1 2 3 4 / \n",
+        "OTHERPROP\n 1 2 3 /\n 4 5 /\n PROP\n 1 2 3 4 / \n",
+        "OTHERPROP\n 1 2 3 /\n 4 5 /\n/ PROP Eclipse comment\n PROP\n 1 2 3 4 / \n",
+        "PROP\n 1 2 3 4 /",
+        "PROP\n -- a comment \n 1 2 3 4 /",
+        "-- a comment \n PROP\n \n 1 2 3 4 /",
+        "PROP\n \n 1 2 \n -- a comment \n 3 4 /",
+        "NOECHO\n PROP\n \n 1 2 \n -- a comment \n 3 4 /",
+        "ECHO\n PROP\n \n 1 2 \n -- a comment \n 3 4 /",
+        "NOECHO\n PROP\n \n 1 2 \n -- a comment \n 3 4 / \n ECHO",
+    ],
+)
+def test_read_simple_property(file_data):
+    with patch("builtins.open", mock_open(read_data=file_data)) as mock_file:
+        with open_grdecl(mock_file, keywords=["PROP"]) as kw:
+            assert list(kw) == [("PROP", ["1", "2", "3", "4"])]
+
+
+def test_read_extra_keyword_characters():
+    file_data = (
+        "LONGPROP Eclipse comment\n"
+        "1 2 3 4 / More Eclipse comment\n OTHERPROP\n 5 6 7 8 /\n"
+    )
+    with patch("builtins.open", mock_open(read_data=file_data)) as mock_file:
+        with open_grdecl(mock_file, keywords=["LONGPROP", "OTHERPROP"]) as kw:
+            assert list(kw) == [
+                ("LONGPROP", ["1", "2", "3", "4"]),
+                ("OTHERPROP", ["5", "6", "7", "8"]),
```

Comment:
Should we also add the keyword `THEVERYLONGKEYWORDTHATISDEFINETIVELYLONGERTHANTHIRTYTWO`? Just to mark that we've solved the initial issue?

Reply:
Sure :)

Reply:
Done